miden-lib/asm/miden/kernels/tx/memory.masm

```diff
@@ -120,6 +120,9 @@ const.ACCT_CORE_DATA_SECTION_END_OFFSET=404
 # The memory address at which the account storage slot type data beings
 const.ACCT_STORAGE_SLOT_TYPE_DATA_OFFSET=405
 
+# The memory address at which the account procedures section begins.
+const.ACCT_PROCEDURES_SECTION_OFFSET=700
```

Comment:
@bobbinth What do you think of 700 as a memory location? 

Considering that there are storage slots before starting at 405, I want to make sure that in the case where we have a large storage for an account  it does not overflow onto the account procedures section.

Reply:
I think this is fine but also we could probably start at 1000 or something like that. There is no penalty for using a higher memory offset and I think the next used one is at 1,048,576 (for the input notes section).

btw, it would be awesome to add a comprehensive comment somewhere at the top of this file (and maybe somewhere else in the docs) explaining the kernel memory layout at the high level. I think currently we have:

| Section          | Start address | End address |
| ------------- | :------------: | :-----------: |
| Bookkeeping | 0                     | 4                    |
| Global inputs | 100                 | 105               |
| Block header | 200                | 207               |
| Chain MMR   | 300                 | up to 332?  |
| Account data | 400               | up to 651?   |
| Account procedures | 700    | ?                  |
| Input notes     | 1_048_576   | ?                  |
| Output notes  | 4_194_304  | ?                  |

Reply:
cc @Fumuran - the table above is relevant for the work you are doing too.

Reply:
I did not find the information for account and chain why 332 and 651 ?

Reply:
For Chain MMR, I believe we can have at most 32 roots - but this needs to be double-checked.

Account data currently also contains slot types for all storage slots (256 elements or 64 words). So, maybe it should be 468 (400 + 4 + 64). But no need to updated this now as it will have to be updated in #802.